### PR TITLE
Unsquish widgets on the Indirect/Inelastic GUI's

### LINKED ISSUE #36382: Investigate squished widgets in Indirect/Inelastic GUIs
**Describe the bug**
I can see when I open an Indirect/Inelastic GUIs there is often at least one widget which is "squished" and hard to read. If no one else can see this, it might be due to the screen resolution on my laptop. For example on the Data Reduction ISIS Energy Transfer:

image

On the diffraction interface:
image

On Data manipulations interface:
image

We would like to investigate why this is happened and fix it before the release of v6.9.

**To Reproduce**
Open one of the interfaces mentioned above

**Expected behavior**
The widgets should not be squished. The widgets should dynamically resize depending on the resolution of your screen.

**Platform/Version (please complete the following information):**
 - OS: Found on Windows

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR fixes a bug where certain widgets on the Indirect/Inelastic were appearing as being "squished" on screens with high resolutions. It was caused by many of the widgets and QGroupBoxes containing the widgets having a minimum and/or maximum vertical and horizontal size set. This is a bad idea because the size is specified in pixels, and so it is not scalable for screens with larger resolutions. It also doesn't scale with the font size inside the widget.

The solution was to remove this manual setting of min and max sizes, and to just let the Qt framework to handle the sizing of widgets for different resolutions.

Fixes #36382

### To test:
It is hard to test this if you do not have a machine with high resolution. Here are screenshots to prove the widgets are no longer squished. Compare these to the images in the attached issue.
Data Reduction - ISIS Calibration
image

Data Manipulation - Iqt Tab
image

Diffraction
image

A code review should be all that is needed.

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by MohamedAlmaki — COMMENTED** [review]
Thanks for the changes, I've noticed this problem for a while. 

There is a problem in some tabs where the fields get squashed when you decrease the height of the tab, for example:
image

**Comment by robertapplin** [comment]
> Thanks for the changes, I've noticed this problem for a while.
> 
> There is a problem in some tabs where the fields get squashed when you decrease the height of the tab

Good spot, this should be fixed in the latest commit

**Comment by MohamedAlmaki** [comment]
Unassinging myself because I will be on leave until the next month

File: docs/source/release/v6.9.0/Inelastic/Bugfixes/36382.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+- A bug where the ``Run`` and ``Ouput Options`` appeared squished on Inelastic interfaces has been fixed.
```
Comment (cailafinn) [inline docs/source/release/v6.9.0/Inelastic/Bugfixes/36382.rst:1]:
```suggestion
- A bug where the ``Run`` and ``Output Options`` appeared squished on Inelastic interfaces has been fixed.
```

**Review by cailafinn — CHANGES_REQUESTED** [review]
Looks good, just typos in the release notes.

**Review by cailafinn — APPROVED** [review]
Looks good to me. Tested on macOS with a Retina Screen (~2k).

**Pull request merged**